src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkCorrectUsingMethodsInspection.kt

```diff
@@ -43,19 +52,21 @@ class SmkCorrectUsingMethodsInspection: SnakemakeInspection() {
                 return
             }
 
+            val firstChildSection = st.firstChild
+
             st.argumentList?.node?.getChildren(null)
                     ?.filter { node -> node.elementType == CALL_EXPRESSION }
                     ?.forEach { node ->
 
                         val method = when (node.firstChildNode.text) {
-                            "ancient" -> checkAncient(st.firstChild)
-                            "protected" -> checkProtected(st.firstChild)
-                            "directory" -> checkDirectory(st.firstChild)
+                            METHOD_ANCIENT -> checkAncient(firstChildSection)
+                            METHOD_PROTECTED -> checkProtected(firstChildSection)
+                            METHOD_DIRECTORY -> checkDirectory(firstChildSection)
```

Comment:
Please check other methods from same source file, e.g. `temp(..)`  and etc

Reply:
I added `temp` and `report` methods, I'm not sure about their usage, should be check

Reply:
You need to be sure to move further, so read the source code/documentation/check runtime behavior.

Reply:
I added next methods `touch`, `pipe`, `repeat`, `unpack`, `dynamic` .
I guided by this documentation and methods description to determine the usage of methods. 
https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html